File: content/snippets/go/interview_lcci/10/11_peaks_and_valleys.go
package main

import (
	"fmt"
	"sort"
)

func wiggleSort(nums []int) {
	sort.Ints(nums)
	for i := 0; i+1 < len(nums); i += 2 {
		nums[i], nums[i+1] = nums[i+1], nums[i]
	}
}

func main() {
	nums := []int{5, 3, 1, 2, 3}
	wiggleSort(nums)
	fmt.Println(nums)
}

File: content/snippets/python/interview_lcci/10/11_peaks_and_valleys.py
from typing import List


def wiggle_sort(nums: List[int]) -> None:
    nums.sort()
    for index in range(0, len(nums) - 1, 2):
        nums[index], nums[index + 1] = nums[index + 1], nums[index]


def main() -> None:
    nums = [5, 3, 1, 2, 3]
    wiggle_sort(nums)
    print(nums)


if __name__ == "__main__":
    main()
